pkg/azdiskschedulerextender/azdiskschedulerextender.go

```diff
@@ -183,38 +186,17 @@ func filter(context context.Context, schedulerExtenderArgs schedulerapi.Extender
 	// filter the nodes based on AzDiverNode status and number of disk attachments
 	failedNodes = make(map[string]string)
 	for _, node := range allNodes {
-		ready, found := nodeNameToAzDriverNodeStatusMap[node.Name]
-		excludeNode := false
-		if found && ready {
-			// TODO: remove maxDiskCount calculations when we fix #1097 in the controller
-			// get AzVolumeAttachments for the node
-			azVolumeAttachments, err := getAdditionalAzVolumeAttachmentsOnNode(node.Name, diskNamesForRequestedVolumes)
-			klog.V(2).Infof(
-				"number of additional volume attachments for node %s and pod %s is %d. Number of requested volumes is %d. Err: %v",
-				node.Name, schedulerExtenderArgs.Pod.Name, len(azVolumeAttachments), len(diskRequestedByPod), err)
-			if err != nil {
-				klog.Warningf("failed to get azVolumeAttachments attached to node (%s): %v", node.Name, err)
-				continue
-			}
-
-			// get the max disk count for the node type
-			maxDiskCount, err := azureutils.GetNodeMaxDiskCountWithLabels(node.Labels)
-			if err != nil {
-				klog.Warningf("failed to get node (%s)'s max disk count: %v", node.Name, err)
-				continue
-			}
-
-			if len(azVolumeAttachments)+totalRequestedDiskCount > maxDiskCount {
-				klog.Warningf("max number of %v AzVolumeAttachments for node %s has been reached. The node will be skipped", maxDiskCount, node.Name, azDriverNodesMeta.err)
-				excludeNode = true
-				failedNodes[node.Name] = "node(s) reached the maximum number of disk attachments"
-			}
-			if !excludeNode {
+		if ready, found := nodeNameToAzDriverNodeStatusMap[node.Name]; found {
+			if ready {
 				filteredNodes = append(filteredNodes, node)
 				storeFilteredNodeNames = append(storeFilteredNodeNames, node.Name)
+			} else {
+				failedNodes[node.Name] = fmt.Sprintf("AzDriverNode for %s is not ready.", node.Name)
 			}
 		} else {
-			failedNodes[node.Name] = fmt.Sprintf("AzDriverNode for %s is not ready.", node.Name)
+			// if AzDriverNode cannot be found for the node, either 1) AzDriverNode cache is stale or 2) the node plugin has not started and needs to be scheduled.
+			// so add the node to filtered node list.
+			storeFilteredNodeNames = append(storeFilteredNodeNames, node.Name)
 		}
 	}
 	filteredNodeNames = &storeFilteredNodeNames
```

Comment:
Should we add the check for empty ` filteredNodeNames`?

Reply:
Please refer to the comment in the group chat. I think as long as the extender can handle the `AzDriverNode` missing case, we should maintain the behavior of returning an empty filtered node list if all nodes in cluster are unhealthy. This should retrigger scheduling until healthy nodes become available. What do you think?

Reply:
The recent update achieves what I was hoping.